src/components/Compare/CompareMain.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { useState, Fragment } from 'react';
+import CompareTable from 'components/Compare/CompareTable';
+import { Modal } from '@material-ui/core';
+import ModalCompare from 'components/Compare/ModalCompare';
+
+const CompareMain = (props: {
+  stateId?: string;
+  stateName?: string;
+  county: any | null;
+  isModal?: Boolean;
+  locationsViewable?: number;
+  isLocationPage?: Boolean;
+  isHomepage?: Boolean;
+}) => {
```

Comment:
Consider making this a bit more generic instead of passing flags to customize the behavior between pages. For example, it seems that the only difference between the location and home pages is the list of locations to show. Maybe we could change it to something like:

```tsx
{/* Homepage */}
<CompareMain
  numLocations={5}
  locations={stateList}
  title='Compare states'
/>

{/* LocationPage */}
<CompareMain 
  currentLocation={okanoganCounty}
  numLocations={5}
  locations={countyList}
  title={isState ? 'Compare counties' : 'compare to other counties'}
/>
```

In the second case, we can pass a location item if we need it to highlight the current location.

Reply:
made progress on this. adding TODO to refactor further and make the CompareMain/CompareTable fully generic